### PR TITLE
Add option to build OP-TEE for non-SMP Linux builds

### PR DESCRIPTION
**Pull request by glneo**
Currently we assume that the shared memory between worlds will be mapped with the same MMU attributes. This is required or cache corruption can result. When compiled without SMP Linux uses cache attributes that do not match what OP-TEE uses, leading to corruption. Add an option to set our MMU attributes to match non-SMP Linux.

A better fix would be to have OP-TEE send its cache attributes over to Linux and have Linux memremap() to the correct attributes. This is far beyond the scope of what is needed right now, this patch seems to be the smallest change set that fixes this issue without effecting other platforms.

### CODE REVIEW

File: core/arch/arm/mm/core_mmu_v7.c [hunk 1]
```diff
@@ -69,9 +69,16 @@
 /* Normal memory, Outer Write-Back Write-Allocate Cacheable */
 #define TEE_MMU_TTB_RNG_WBWA    (1 << 3)
 
+/* Normal memory, Outer Write-Back no Write-Allocate Cacheable */
+#define TEE_MMU_TTB_RNG_WB      (3 << 3)
+
+#ifndef CFG_NO_SMP
```
Comment (jenswi-linaro) [inline core/arch/arm/mm/core_mmu_v7.c]: Could we use `CFG_TEE_CORE_NB_CORE` instead or is it a valid use case to have one core, but still in SMP configuration?

Wouldn't `TTBCR`, `TCR_EL1` need to be update also?

Comment (glneo) [inline core/arch/arm/mm/core_mmu_v7.c]: It is valid to have one core in SMP, we do this on some of our platforms for coherency reasons with non-ARM cores.
TTBCR should be fine, and this is only a fix for ARMv7.

Comment (jenswi-linaro) [inline core/arch/arm/mm/core_mmu_v7.c]: Please fix ARMv8 and ARMv7 with LPAE also, or at least introduce a compile error if `CFG_NO_SMP` is set.

Comment (glneo) [inline core/arch/arm/mm/core_mmu_v7.c]: This fix is only relevant for that ARMv7 without LPAE, this issue does not seem to affect LPAE or ARMv8.

Comment (jenswi-linaro) [inline core/arch/arm/mm/core_mmu_v7.c]: Please clarify, how does it not affect LPAE or ARMv8?

Comment (glneo) [inline core/arch/arm/mm/core_mmu_v7.c]: Linux with LPAE uses different attributes which seem to match OP-TEE, I have not done extensive analysis, but testing on our LPAE platforms seems to indicate this, I do not have an ARMv8 platform to test but it should be the same case as for LPAE.

Comment (jenswi-linaro) [inline core/arch/arm/mm/core_mmu_v7.c]: OK

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by glneo** [comment]
Updated

**Pull request merged**